tests/neo4j/sessionrun.py

```diff
@@ -207,6 +207,54 @@ def test_iteration_larger_than_fetch_size(self):
             rec = result.next()
             self.assertEqual(rec, exp)
 
+    def test_partial_iteration(self):
```

Comment:
This is a good point and I actually thought about splitting it. The thing is, that these integration tests are rather costly in terms of runtime. And if I split them, I'll have to send 4 instead of 3 queries to the server + whatever fluff is needed around it (e.g., TestKit protocol communication, etc).